xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -463,3 +483,105 @@ func circuitBreakersFromCluster(cluster *v3clusterpb.Cluster) *uint32 {
 	}
 	return nil
 }
+
+// outlierConfigFromCluster extracts the relevant outlier detection configuration from
+// the received cluster resource. Returns nil if no OutlierDetection.
+func outlierConfigFromCluster(cluster *v3clusterpb.Cluster) (*OutlierDetection, error) {
+	// Corresponds to this line from A50: "If the outlier_detection field is not
+	// set in the Cluster message, a "no-op" outlier_detection config will be
+	// generated"
+	od := cluster.GetOutlierDetection()
+	if od == nil {
+		return nil, nil
+	}
+	// "The google.protobuf.Duration fields interval, base_ejection_time, and
+	// max_ejection_time must obey the restrictions in the
+	// google.protobuf.Duration documentation and they must have non-negative
+	// values." - A50
+	// TODO: I can find non negative examples for validation, but not "must obey the restrictions in the documentation"
```

Comment:
Can you check if we have some tests for other fields which are of type `google.protobuf.Duration`. I also vaguely remember it being a problem to pass invalid values for these fields in tests since JSON unmarshaling would fail or something like that.

Reply:
Also there are a couple of methods on the `Duration` type: `IsValid()` and `CheckValid()`. They might be handy here.

Reply:
Ah, awesome, thank you so much. I decided to have a separate conditional and call CheckValid and return error for logging. Let me know if you think both validations/errors should be combine into one.

Reply:
Added tests for all three duration fields that trigger CheckValid to return a non-nil error.

Reply:
Yeah, whoops. Deleted todo.